Library/Homebrew/rubocops/shell_cmd.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits shell commands `system` and `Utils.popen`.
+      class ShellCmd < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          popen_commands = [
+            :popen,
+            :popen_read,
+            :safe_popen_read,
+            :popen_write,
+            :safe_popen_write,
+          ]
+
+          popen_commands.each do |popen_command|
+            find_every_method_call_by_name(body_node, popen_command).each do |method|
+              next unless match = regex_match_group(parameters(method).first, /([^"' ]+)=([^"' ]+) ([^"']+)/)
+
+              good_args = "Utils.#{popen_command}({ \"#{match[1]}\" => \"#{match[2]}\" }, \"#{match[3]}\")"
+              problem "Use `#{good_args}` instead of `Utils.#{popen_command} \"#{match[0]}\"`"
```

Comment:
I'm not yet sure how well this will work in tandem with #7691

Two concerns at the moment:
1. The regex in #7691 will not catch this issue: `Utils.popen_read({ "SHELL" => "bash" }, "foo bar")`
1. This regex will not catch this issue: `Utils.popen_read "SHELL=bash", "foo", "bar"`